plugin/etcd/plugin.go

```diff
@@ -0,0 +1,451 @@
+				Mode:    "target",
+				Name:    "auth",
+				Type:    "bool",
+				Title:   "Authentication",
+				Help:    "Is role based or cert based authentication enabled on the etcd cluster",
+				Example: "false",
+			},
+			plugin.Field{
+				Mode:    "target",
+				Name:    "user",
+				Type:    "string",
+				Help:    "Username for role based authentication",
+				Title:   "Username",
+				Example: "admin",
+			},
+			plugin.Field{
+				Mode:    "target",
+				Name:    "pass",
+				Type:    "password",
+				Help:    "Password for role based authentication",
+				Title:   "Password",
+				Example: "p@ssw0rd",
+			},
+			plugin.Field{
+				Mode:    "target",
+				Name:    "clientCertPath",
+				Type:    "string",
+				Help:    "Path to the certificate issued by the CA for the client connecting to the ETCD cluster",
+				Title:   "Client Certificate File Path",
+				Example: "/tmp/test-certs/test-client.crt",
+			},
+			plugin.Field{
+				Mode:    "target",
+				Name:    "clientKeyPath",
+				Type:    "string",
+				Help:    "Path to the key issued by the CA for the client connecting to the ETCD cluster",
+				Title:   "Client Key File Path",
+				Example: "/tmp/test-certs/test-client-key.key",
+			},
+			plugin.Field{
+				Mode:    "target",
+				Name:    "caCertPath",
+				Type:    "string",
+				Help:    "Path to the CA certificate that issued the client cert and key",
+				Title:   "Trusted CA File Path",
+				Example: "/tmp/test-certs/test-ca.crt",
+			},
+			plugin.Field{
+				Mode:    "target",
+				Name:    "fullOverwrite",
+				Type:    "bool",
+				Help:    "If this is enabled, the existing key/value pairs will be deleted. The values will be restored using the backup archive.",
+				Title:   "Full Overwrite",
+				Example: "false",
+			},
+			plugin.Field{
+				Mode:    "target",
+				Name:    "prefix",
+				Type:    "string",
+				Help:    "This is the input string for prefix based backup-restore",
+				Title:   "Prefix",
+				Example: "starkandwayne/",
+			},
+		},
+	}
+	fmt.Fprintf(os.Stderr, "etcd plugin starting up...\n")
+	plugin.Run(p)
+}
+
+type EtcdPlugin plugin.PluginInfo
+
+type EtcdConfig struct {
+	EtcdEndpoints  string
+	Authentication bool
+	Username       string
+	Password       string
+	ClientCertPath string
+	ClientKeyPath  string
+	CaCertPath     string
+	FullOverwrite  bool
+	Prefix         string
+}
+
+func (p EtcdPlugin) Meta() plugin.PluginInfo {
+	return plugin.PluginInfo(p)
+}
+
+func getEtcdConfig(endpoint plugin.ShieldEndpoint) (*EtcdConfig, error) {
+	etcdEndpoint, err := endpoint.StringValue("endpoints")
+	if err != nil {
+		return nil, err
+	}
+
+	auth, err := endpoint.BooleanValueDefault("auth", false)
+	if err != nil {
+		return nil, err
+	}
+
+	user, err := endpoint.StringValueDefault("user", "")
+	if err != nil {
+		return nil, err
+	}
+
+	pass, err := endpoint.StringValueDefault("pass", "")
+	if err != nil {
+		return nil, err
+	}
+
+	clientCert, err := endpoint.StringValueDefault("clientCertPath", "")
```

Comment:
If X.509 certificates are required, based on auth mode, you should ensure that they are valid here, and not use `StringValueDefault()`.